spyder/app/mainwindow.py

```diff
@@ -1677,25 +1693,38 @@ def setup_default_layouts(self, index, settings):
                                      Qt.Horizontal)
 
         # Arrange rows vertically
-        for column in widgets_layout :
+        for column in widgets_layout_clean:
             for i in range(len(column) - 1):
                 first_row, second_row = column[i], column[i+1]
-                if first_row is not None and second_row is not None:
-                    self.splitDockWidget(first_row[0].dockwidget,
-                                         second_row[0].dockwidget,
-                                         Qt.Vertical)
+                self.splitDockWidget(first_row[0].dockwidget,
+                                     second_row[0].dockwidget,
+                                     Qt.Vertical)
         # Tabify
-        for column in widgets_layout :
+        for column in widgets_layout_clean:
             for row in column:
                 for i in range(len(row) - 1):
                     first, second = row[i], row[i+1]
-                    if first is not None and second is not None:
-                        self.tabify_plugins(first, second)
+                    self.tabify_plugins(first, second)
 
                 # Raise front widget per row
                 row[0].dockwidget.show()
                 row[0].dockwidget.raise_()
 
+        # Set widths and heights
+        width_fractions = layout['width fraction']
+        if len(width_fractions) > 1:
+            # TODO: get the first col[i] that is not None
+            _widgets = [col[0][0].dockwidget for col in widgets_layout]
+            print(_widgets, width_fractions, Qt.Horizontal)
+            self.resizeDocks(_widgets, width_fractions, Qt.Horizontal)
+
+        height_fractions = layout['height fraction']
+        for i, column in enumerate(widgets_layout_clean):

```

Comment:
``idx`` seems to be the modern convention

Reply:
Agreed ;-)